QRCoder/BitmapByteQRCode.cs

```diff
@@ -53,13 +53,35 @@ public byte[] GetGraphic(int pixelsPerModule, byte[] darkColorRgb, byte[] lightC
     {
         var sideLength = QrCodeData.ModuleMatrix.Count * pixelsPerModule;
 
-        var moduleDark = darkColorRgb.Reverse();
-        var moduleLight = lightColorRgb.Reverse();
+        var moduleDarkSingle = darkColorRgb.Reverse().ToArray();
+        var moduleLightSingle = lightColorRgb.Reverse().ToArray();
 
-        var bmp = new List<byte>();
+        // Pre-calculate color/module bytes
+        var moduleDark = new byte[pixelsPerModule * 3];
+        var moduleLight = new byte[pixelsPerModule * 3];
+        for (int i = 0; i < pixelsPerModule; i++)
+        {
+            Array.Copy(moduleDarkSingle, 0, moduleDark, i * 3, 3);
+            Array.Copy(moduleLightSingle, 0, moduleLight, i * 3, 3);
+        }
+
+        // Pre-calculate padding bytes
+        var padding = new byte[sideLength % 4];
+
+        // Calculate filesize (header + pixel data + padding)
+        var fileSize = 54 + (3 * (sideLength * sideLength)) + (sideLength * padding.Length);
+
+
+        var bmp = new List<byte>(fileSize);
+
+        //header part 1
+        bmp.AddRange(new byte[] { 0x42, 0x4D });
```

Comment:
This is an allocation; these headers could be placed in a readonly static variable and copied from there.